### PR TITLE
server,pgwire: make more log message bits non-redactable

### PR DESCRIPTION
**Pull request by knz**
Commit prefix from  #53764
see individual commits for details

Release justification: low risk, high benefit changes to existing functionality

### CODE REVIEW

**Comment by knz** [comment]
cc @spaskob @lucy-zhang to review SQL schema and jobs adjustments

**Comment by knz** [comment]
This work should be improved once https://github.com/cockroachdb/redact/issues/2 is solved.

**Review by spaskob — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53312#-:-MFf5wIS5WerBFdWMWfH:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @irfansharif)

File: pkg/cli/start.go [hunk 1]
```diff
@@ -607,9 +608,10 @@ If problems persist, please see %s.`
 			var buf bytes.Buffer
 			info := build.GetInfo()
 			tw := tabwriter.NewWriter(&buf, 2, 1, 2, ' ', 0)
-			fmt.Fprintf(tw, "CockroachDB node starting at %s (took %0.1fs)\n", timeutil.Now(), timeutil.Since(tBegin).Seconds())
-			fmt.Fprintf(tw, "build:\t%s %s @ %s (%s)\n", info.Distribution, info.Tag, info.Time, info.GoVersion)
-			fmt.Fprintf(tw, "webui:\t%s\n", serverCfg.AdminURL())
+			redact.Fprintf(tw, "CockroachDB node starting at %s (took %0.1fs)\n", timeutil.Now(), timeutil.Since(tBegin).Seconds())
```
Comment (irfansharif) [inline pkg/cli/start.go:611]: I'm assuming we're going to do something here to make this less of a footgun? (was discussed yesterday in the team meeting).

**Review by irfansharif — APPROVED** [review]
LGTM, hopefully this was relaxing work.

File: pkg/cli/start.go (hunk 1 above)
Comment (knz) [reply pkg/cli/start.go:611]: Yes once https://github.com/cockroachdb/redact/pull/4 is merged I will rebase on top of that and remove the footgun here.

File: docs/generated/redact_safe.md [hunk 2]
```diff
@@ -10,6 +10,9 @@ pkg/roachpb/metadata.go | `RangeID`
 pkg/roachpb/metadata.go | `ReplicaID`
 pkg/roachpb/metadata.go | `RangeGeneration`
 pkg/roachpb/metadata.go | `ReplicaType`
+pkg/sql/catalog/descpb/structured.go | `ID`
```
Comment (lucy-zhang) [inline docs/generated/redact_safe.md:14]: Thanks. Is it possible to also add some of the other descriptor IDs in `descpb/structured.go` (`ColumnID`, `IndexID`, `FamilyID`, `MutationID`)?

Comment (knz) [reply docs/generated/redact_safe.md:14]: taking note, will do

Comment (lucy-zhang) [reply docs/generated/redact_safe.md:14]: Thanks. I think we want `descpb.DescriptorVersion` as well.

**Review by knz — COMMENTED** [review]
> Thanks. I think we want descpb.DescriptorVersion as well.

Done.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53312#-:-MGDSb9rAi8hpjihaO9p:b-8z8fsz)* status: :shipit: complete! 1 of 0 LGTMs obtained

File: pkg/cli/start.go (hunk 1 above)
Comment (knz) [reply pkg/cli/start.go]: Done.

**Comment by knz** [comment]
tfyrs

bors r=spaskob,irfansharif

**Pull request merged**